src/dune_rules/dir_with_dune.ml

```diff
@@ -22,3 +22,20 @@ and inner_fold t inner_list l ~init ~f =
   match inner_list with
   | [] -> deep_fold l ~init ~f
   | x :: inner_list -> inner_fold t inner_list l ~init:(f t x init) ~f
+
+module Deep_fold (M : Monad_intf.S) = struct
```

Comment:
This module is not exposed and used only once. Did you mean to apply to `Monad.Id` to factorise the code above?

Reply:
Indeed. I forgot to get back to this and get remove the non monadic impl.